File: Javascript/CodingGame/Telephone_numbers.js
/**
 * https://www.codingame.com/ide/puzzle/telephone-numbers
 **/

 const N        = parseInt(readline());
 const elements = new Set()
 
 for (let i = 0; i < N; i++) {
     const telephone = readline();
 
     for (let y = 0; y < telephone.length; y++) {
         elements.add(telephone.substring(0, y + 1));
     }
 }
 
 console.log(elements.size);

File: Python/CodingGame/Telephone_numbers.py
# https://www.codingame.com/ide/puzzle/telephone-numbers

n = int(input())
elements = set()

for _ in range(n):
    telephone = input()

    for i in range(len(telephone)):
        elements.add(telephone[:i + 1])

# The number of elements (referencing a number) stored in the structure.
print(len(elements))
